app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt

```diff
@@ -165,25 +170,51 @@ class BrowserFragment : IWebViewLifecycleFragment() {
             onNavigationEvent = this@BrowserFragment.onNavigationEvent
             navigationStateProvider = NavigationStateProvider()
             visibility = overlayVisibleCached ?: View.GONE
+            openHomeTileContextMenu = {
+                activity.openContextMenu(browserOverlay.tileContainer)
+            }
             onPreSetVisibilityListener = { isVisible ->
                 // The overlay can clear the DOM and a previous focused element cache (e.g. reload)
                 // so we need to do our own caching: see FocusedDOMElementCache for details.
                 if (!isVisible) { webView?.focusedDOMElement?.cache() }
             }
+            registerForContextMenu(browserOverlay.tileContainer)
         }
 
         layout.progressBar.initialize(this)
 
         return layout
     }
 
+    override fun onContextItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.remove -> {
+                val homeTileAdapter = tileContainer.adapter as HomeTileAdapter
+                val tileToRemove = homeTileAdapter.getItemAtPosition(browserOverlay.getFocusedTilePosition())
+                        ?: return false
+                // This assumes that since we're deleting from a Home Tile object that we created
+                // that the Uri is valid, so we do not do error handling here.
+                HomeTilesManager.removeHomeTile(tileToRemove, context)
+                homeTileAdapter.removeItemAtPosition(browserOverlay.getFocusedTilePosition())
+                TelemetryWrapper.homeTileRemovedEvent(tileToRemove)
+                return true
+            }
+            else -> return false
+        }
+    }
+
     override fun onDestroyView() {
         super.onDestroyView()
         lifecycle.removeObserver(cursor!!)
         cursor = null
         overlayVisibleCached = browserOverlay.visibility
     }
 
+    override fun onCreateContextMenu(menu: ContextMenu?, v: View?, menuInfo: ContextMenu.ContextMenuInfo?) {
+        super.onCreateContextMenu(menu, v, menuInfo)
```

Comment:
You don't want to call super: it will call `Activity.onCreateContextMenu`. Check the docs